File: data_v4/ale-04/verify/baseline.py
#!/usr/bin/env python3
"""Trivial baseline: 'honor every target' coloring.

Each cell outputs its pinned value if pinned, else its target t[i]. This is always
FEASIBLE (pins honored, all bits) but pays a large interface penalty along every
target blob boundary -- the coloring the heuristic must beat.

Usage: python3 baseline.py <instance_file>   (writes solution grid to stdout)
"""
import sys


def main():
    with open(sys.argv[1]) as f:
        toks = f.read().split()
    it = iter(toks)
    N = int(next(it)); _W = int(next(it))
    rows = []
    for r in range(N):
        row = []
        for c in range(N):
            _h = int(next(it)); t = int(next(it)); p = int(next(it))
            row.append(str(p if p != -1 else t))
        rows.append(" ".join(row))
    sys.stdout.write("\n".join(rows) + "\n")


if __name__ == "__main__":
    main()

File: data_v4/ale-v2-05/verify/baseline.cpp
// Trivial baseline for ale-v2-05 (UFLP): open ALL facilities.
// Always feasible (M = F >= 1, distinct indices). Service cost is minimal but
// it pays every opening cost, so it is a weak-but-legal reference to beat.
#include <bits/stdc++.h>
using namespace std;
int main(){
    int F, C;
    if(!(cin>>F>>C)) return 0;
    for(int i=0;i<F;i++){long long x,y,c;cin>>x>>y>>c;}
    for(int i=0;i<C;i++){long long x,y;cin>>x>>y;}
    cout<<F<<"\n";
    for(int i=0;i<F;i++) cout<<(i+1)<<"\n";
    return 0;
}
